src/command.rs

```diff
@@ -573,3 +577,169 @@ impl Command for Echo {
+    type Output = bool;
+
+    fn into_request(self) -> RespValue {
+        RespValue::Array(vec![
+            "SCRIPT".into(),
+            "EXISTS".into(),
+            RespValue::BulkString(self.hash),
+        ])
+    }
+
+    fn from_response(res: RespValue) -> Result<Self::Output, RespError> {
+        match res {
+            RespValue::Integer(0) => Ok(false),
+            RespValue::Integer(1) => Ok(true),
+            res => Err(RespError::RESP(
+                "invalid response for SCRIPT EXISTS".into(),
+                Some(res),
+            )),
+        }
+    }
+
+    fn key_slot(&self) -> Result<Option<u16>, Vec<u16>> {
+        Ok(Some(self.slot))
+    }
+}
+
+pub struct ScriptLoad<'a> {
+    pub script: &'a str,
+    pub slot: u16,
+}
+
+impl<'a> Message for ScriptLoad<'a> {
+    type Result = Result<Vec<u8>, Error>;
+}
+
+impl<'a> Command for ScriptLoad<'a> {
+    type Output = Vec<u8>;
+
+    fn into_request(self) -> RespValue {
+        resp_array!["SCRIPT", "LOAD", self.script]
+    }
+
+    fn from_response(res: RespValue) -> Result<Self::Output, RespError> {
+        match res {
+            RespValue::BulkString(hash) => Ok(hash),
+            res => Err(RespError::RESP(
+                "invalid response for SCRIPT LOAD".into(),
+                Some(res),
+            )),
+        }
+    }
+
+    fn key_slot(&self) -> Result<Option<u16>, Vec<u16>> {
+        Ok(Some(self.slot))
+    }
+}
+
+pub struct ScriptFlush {
+    pub slot: u16,
+}
+
+impl Message for ScriptFlush {
+    type Result = Result<String, Error>;
+}
+
+impl Command for ScriptFlush {
+    type Output = String;
+    fn into_request(self) -> RespValue {
+        resp_array!["SCRIPT", "FLUSH"]
+    }
+
+    fn from_response(res: RespValue) -> Result<Self::Output, RespError> {
+        match res {
+            RespValue::SimpleString(str) => Ok(str),
+            res => Err(RespError::RESP(
+                "invalid response for SCRIPT FLUSH".into(),
+                Some(res),
+            )),
+        }
+    }
+
+    fn key_slot(&self) -> Result<Option<u16>, Vec<u16>> {
+        Ok(Some(self.slot))
+    }
+}
+
+pub struct Eval<'a> {
+    pub script: &'a str,
+    pub keys: Vec<String>,
+    pub args: Vec<RespValue>,
+}
+
+impl<'a> Message for Eval<'a> {
+    type Result = Result<RespValue, Error>;
+}
+
+impl<'a> Command for Eval<'a> {
+    type Output = RespValue;
+
+    fn into_request(mut self) -> RespValue {
+        let mut v = vec![
+            "EVAL".into(),
+            self.script.into(),
+            format!("{}", self.keys.len()).into(),
+        ];
+        for key in self.keys {
+            v.push(key.into());
+        }
+        v.append(&mut self.args);
+        RespValue::Array(v)
+    }
+
+    fn from_response(res: RespValue) -> Result<Self::Output, RespError> {
+        Ok(res)
+    }
+
+    fn key_slot(&self) -> Result<Option<u16>, Vec<u16>> {
+        hash_slot_many(&self.keys)
+    }
+}
+
+pub struct EvalSHA {
```

Comment:
Could you rename it to `EvalSha`?
Rust API Guidelines suggests not capitalizing letters in an acronym: https://rust-lang-nursery.github.io/api-guidelines/naming.html
Other crates follow this convention like [this](https://docs.rs/sha2/0.8.0/sha2/).